src/art.rs

```diff
@@ -579,9 +674,10 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
 
     #[inline]
     pub(crate) fn get_leaf_by_query(&self, query_type: QueryType) -> Option<Arc<LeafValue<V>>> {
-        // Unwrap the NodeType::Twig to access the TwigNode instance.
-        let NodeType::Twig(twig) = &self.node_type else {
-            return None;
+        let twig = if let NodeType::Twig(twig) = &self.node_type {
```

Comment:
Can we document this? It will be easier to understand why we are getting inner_twig here

Another case of why your initial leaf name was much better (my bad for suggesting inner_twig)